src/Authoring/WinRT.Host/WinRT.Host.cpp

```diff
@@ -43,52 +46,68 @@ static load_assembly_and_get_function_pointer_fn load_assembly_and_get_function_
-        { ResolverResolveFailure, L"Resolution of dependencies in hostpolicy failed." },
-        { LibHostCurExeFindFailure, L"Failure to determine the location of the current executable." },
-        { LibHostInitFailure, L"Initialization of the hostpolicy library failed." },
-        { LibHostSdkFindFailure, L"Failure to find the requested SDK. This happens in the hostfxr when an SDK(also called CLI) command is used with dotnet. In this case the hosting layer tries to find an installed. NET SDK to run the command on. The search is based on deduced install locationand on the requested version from potential global. json file. If either no matching SDK version can be found, or that version exists, but it's missing the dotnet. dll file, this error code is returned." },
-        { LibHostInvalidArgs, L"Arguments to hostpolicy are invalid." },
-        { InvalidConfigFile, L"The .runtimeconfig.json file is invalid." },
-        { AppArgNotRunnable, L"The command line for dotnet.exe doesn't contain the path to the application." },
-        { AppHostExeNotBoundFailure, L"Apphost failed to determine which application to run." },
-        { FrameworkMissingFailure, L"It was not possible to find a compatible framework version." },
-        { HostApiFailed, L"The hostpolicy could not calculate the NATIVE_DLL_SEARCH_DIRECTORIES." },
-        { HostApiBufferTooSmall, L"The buffer specified to an API is not big enough to fit the requested value." },
-        { LibHostUnknownCommand, L"The corehost_main_with_output_buffer is called with unsupported host command." },
-        { LibHostAppRootFindFailure, L"The imprinted application path doesn't exist." },
-        { SdkResolverResolveFailure, L"hostfxr_resolve_sdk2 failed to find matching SDK." },
-        { FrameworkCompatFailure, L"The .runtimeconfig.json contains two incompatible framework references." },
-        { FrameworkCompatRetry, L"A previously processed framework reference was reprocessed." },
-        { BundleExtractionFailure, L"Error extracting single-file apphost bundle." },
-        { BundleExtractionIOError, L"Error reading or writing files during single-file apphost bundle extraction." },
-        { LibHostDuplicateProperty, L"The .runtimeconfig.json contains a runtime property which is also produced by the hosting layer." },
-        { HostApiUnsupportedVersion, L"Feature which requires certain version of the hosting layer binaries was used on a version which doesn't support it." },
-        { HostInvalidState, L"The current state is incompatible with the requested operation." },
-        { HostPropertyNotFound, L"A property requested by hostfxr_get_runtime_property_value doesn't exist." },
-        { CoreHostIncompatibleConfig, L"The component being initialized requires framework which is not available" },
-        { HostApiUnsupportedScenario, L"hostfxr doesn't currently support requesting the given delegate type using the given context." },
+        { InvalidArgFailure, 106 },
```

Comment:
The overloading of the constants with different values is very confusing. At a glance I thought this table was mapping from "106 -> 106" for example. Consider differentiating the resource names with a prefix / suffix in resource.h. e.g.

InvalidArgFailure - > InvalidArgFailureStr

Since they're 100% consistent with the original error names, you could macro-ize the mapping or otherwise simplify the code.

@define CSWINR_RESSTR(x) { x, x##Str },